app/models/concerns/arbitrary_period_comparison_view.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module ArbitraryPeriodComparisonView
```

Comment:
I've [moved some of these methods](https://github.com/Energy-Sparks/energy-sparks/pull/3521/files#diff-2f77ae14cf68c21201b5b2d7044091025c1b9ab86293237ae4a7d3f6be359a94) to `View` base class in my report PR with some tidying and comments. I can rework after this has merged.